src/components/kalman_component/kalman.sh

```diff
@@ -0,0 +1,151 @@
+
+    if ("$DoJacobian" && "$DoInversion" && "$DoPosterior"); then
+
+        # First run the Preview if necessary to get prior emissions
+        # needed for prepare_sf.py
+        if [[ ! -d ${RunDirs}/preview_run/OutputDir ]]; then
+            printf "\nPreview Dir not detected. Running the IMI Preview as a prerequisite for Kalman Mode.\n"
+            run_preview
+        fi
+        # Key directories
+        JacobianRunsDir="${RunDirs}/jacobian_runs"
+        PriorRunDir="${JacobianRunsDir}/${RunName}_0000"
+        PosteriorRunDir="${RunDirs}/posterior_run"
+        InversionDir="${RunDirs}/inversion_template"
+
+        PeriodsFile="${RunDirs}/periods.csv"
+        nPeriods=$(($(wc -l < ${PeriodsFile}) - 1))
+
+        # by default, start with period 1
+        if [[ "x${FirstPeriod}" == "x" ]]; then
+            FirstPeriod=1
+        fi
+        # run inversion for each period
+        for ((i=FirstPeriod;i<=nPeriods;i++)); do
+            run_period
+        done
+    else
+        printf "Exiting: DoJacobian, DoInversion, and DoPosterior must all be set to true to continue with KalmanMode\n"
+    fi
+
+}
+
+# Description: Run inversion for period i
+# Usage:
+#   run_period
+run_period() {
+    ##=======================================================================
+    ##  Setup (dates, emission scale factors)
+    ##=======================================================================
+
+    # Print current period
+    echo -e "\nPeriod ${i}"
+
+    # Create inversion directory for the period
+    cp -r ${RunDirs}/inversion_template/. ${RunDirs}/kf_inversions/period${i}
+
+    # Get Start/End dates of current period from periods.csv
+    ithLine=$(sed "$((i+1))q;d" $PeriodsFile)
+    ithDates=(${ithLine//,/ })
+    StartDate_i=${ithDates[0]}
+    EndDate_i=${ithDates[1]}
+    echo "Start, End: $StartDate_i, $EndDate_i"
+
+    # Set dates in geoschem_config.yml for prior, perturbation, and posterior runs
+    python ${InversionPath}/src/components/kalman_component/change_dates.py $StartDate_i $EndDate_i $JacobianRunsDir; wait
+    python ${InversionPath}/src/components/kalman_component/change_dates.py $StartDate_i $EndDate_i $PosteriorRunDir; wait
+    echo "Edited Start/End dates in geoschem_config.yml for prior/perturbed/posterior simulations: $StartDate_i to $EndDate_i"
+
+    # Prepare initial (prior) emission scale factors for the current period
+    ConfigPath=${InversionPath}/${ConfigFile}
+    echo "python path = $PYTHONPATH"
+    python ${InversionPath}/src/components/kalman_component/prepare_sf.py $ConfigPath $i ${RunDirs} $NudgeFactor; wait
+
+    ##=======================================================================
+    ##  Submit all Jacobian simulations OR submit only the Prior simulation
+    ##=======================================================================
+    
+    # run jacobian simulation for the given period
+    run_jacobian
+
+    # run inversion for the given period
+    run_inversion
```

Comment:
I must be forgetting how the components (including preview, which I commented on above) can see each other from their different directories.

Reply:
see my comment above